### PR TITLE
adds goreleaser support

### PR DESCRIPTION
**Pull request by catherinejones**
This change adds support for goreleaser, which can be called with `goreleaser` or `make release`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,25 +1,22 @@
 # Go parameters
 GOCMD=go
 GODEP=dep
+GORELEASER=goreleaser
```
Comment (klautcomputing) [inline Makefile]: Do we want to have a `make deps` that checks/installs things for you? Checks for go and docker and installs/updates goreleaser and dep?

File: cmd/voucher_cli/main.go [hunk 2]
```diff
@@ -1,5 +1,12 @@
 package main
 
+var (
+	version = "dev"
+	commit  = "none"
+	date    = "unknown"
+)
+
 func main() {
+	rootCmd.Version = version
```
Comment (klautcomputing) [inline cmd/voucher_cli/main.go]: What are we doing with this set version? Does cobra have an inbuilt version command?

File: .gitignore [hunk 3]
```diff
@@ -12,10 +12,7 @@
 *.out
 
 # voucher specific output
-voucher_client
```
[the remark this answers is not in the document]
Comment (catherinejones) [reply .gitignore]: They were also ignoring the contents of `cmd/voucher_{cli,client,server}` and I couldn’t find a simpler way of blocking the binaries without moving them to a different directory.

File: Makefile (hunk 1 above)
Comment (catherinejones) [reply Makefile]: Just using `go get`? And then use the `$GOPATH` to find the binaries?

File: cmd/voucher_cli/main.go (hunk 2 above)
Comment (catherinejones) [reply cmd/voucher_cli/main.go]: It does! If you set the version it automatically enables it!

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (klautcomputing) [reply .gitignore]: Let's make a new dir then! 🙂 `/exe(cutables)`?

File: Makefile (hunk 1 above)
Comment (klautcomputing) [reply Makefile]: I thought something like:
```
DOCKER := $(shell command -v dot 2> /dev/null)

deps:
ifndef DOCKER
error "Docker not installed"
endif
```
and for the other ones either install or update. How does that sound to you?

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (catherinejones) [reply .gitignore]: What about `/build`? Since we already have `/dist` as a result of this change!

File: Makefile (hunk 1 above)
Comment (catherinejones) [reply Makefile]: We could definitely do that! I can add that to this pull request?

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (catherinejones) [reply .gitignore]: Added!

File: Makefile (hunk 1 above)
Comment (catherinejones) [reply Makefile]: I've added `make system-deps` for checking installed dependencies (since we have `make update-deps` for updating with `go dep`)!

**Pull request merged**